extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -1469,16 +1483,15 @@ class LifecycleTask extends BackbeatTask {
                 return done();
             }
             if (rules.Transition) {
-                this._applyTransitionRule({
+                return this._applyTransitionRule({
```

Comment:
While I believe it's a beneficial concept to implement asynchronous logic for better flow control, there's a possibility that it could introduce a delay in the execution of the bucket processor task, as it would need to wait for the callback to complete, potentially affecting the Kafka consumer (as discussed earlier).

Reply:
This could clarify why the processing of Lifecycle bucket tasks doesn't seem very asynchronous – we weren't waiting for the callback to be invoked. This might be due to concerns about its potential impact on the Kafka consumer.

Reply:
individual objects are processed in parallel (hard-coded to 10 at a time), so it should be fine : and if this introduces a delay, we will need to _manage_ it, not just run uncontrolled number of operations in parallel.
Another hint is that the callback is used in lifecycle v2 task.